src/components/tabController/FadedScrollView.tsx

```diff
@@ -0,0 +1,67 @@
+import React, {useCallback} from 'react';
+import {ViewProps, ScrollView, ScrollViewProps, NativeSyntheticEvent, NativeScrollEvent} from 'react-native';
+import Fader from '../fader';
+import withScrollEnabler, {WithScrollEnablerProps} from '../../commons/withScrollEnabler';
+import withScrollReached, {WithScrollReachedProps} from '../../commons/withScrollReached';
+import forwardRef, {ForwardRefInjectedProps} from '../../commons/forwardRef';
+import {Constants} from '../../helpers';
+
+export type FadedScrollViewProps = ViewProps &
+  ScrollViewProps & {
+    children?: React.ReactNode | React.ReactNode[];
+  };
+
+type ScrollReachedProps = FadedScrollViewProps & WithScrollReachedProps;
+type ScrollEnabledProps = ScrollReachedProps & WithScrollEnablerProps;
+type Props = ScrollEnabledProps & ForwardRefInjectedProps;
+
+const FADER_SIZE = 76;
+
+const FadedScrollView = (props: Props) => {
+  const {scrollEnablerProps, scrollReachedProps, children, onScroll: propsOnScroll, ...other} = props;
+  const showLeft =
+    scrollEnablerProps.scrollEnabled &&
+    (Constants.isRTL ? !scrollReachedProps.isScrollAtEnd : !scrollReachedProps.isScrollAtStart);
+  const leftPosition = Constants.isRTL ? Fader.position.RIGHT : Fader.position.LEFT;
+  const showRight =
+    scrollEnablerProps.scrollEnabled &&
+    (Constants.isRTL ? !scrollReachedProps.isScrollAtStart : !scrollReachedProps.isScrollAtEnd);
+  const rightPosition = Constants.isRTL ? Fader.position.LEFT : Fader.position.RIGHT;
+
+  const onScroll = useCallback((event: NativeSyntheticEvent<NativeScrollEvent>) => {
+    scrollReachedProps.onScroll(event);
+      propsOnScroll?.(event);
+  },
+  [scrollReachedProps.onScroll, propsOnScroll]);
+
+  if (children) {
+    return (
+      <>
+        <ScrollView
+          horizontal
+          showsHorizontalScrollIndicator={false}
+          scrollEventThrottle={16}
+          decelerationRate={'fast'}
+          {...other}
+          scrollEnabled={scrollEnablerProps.scrollEnabled}
+          onContentSizeChange={scrollEnablerProps.onContentSizeChange}
+          onLayout={scrollEnablerProps.onLayout}
+          onScroll={onScroll}
+          ref={props.forwardedRef}
+        >
+          {children}
+        </ScrollView>
+        <Fader visible={showLeft} position={leftPosition} size={FADER_SIZE} supportRTL/>
+        <Fader visible={showRight} position={rightPosition} size={FADER_SIZE} supportRTL/>
+      </>
+    );
+  }
+
+  return null;
+};
+
+// TODO: fix withScrollEnabler props (add <>)
+export default withScrollReached<FadedScrollViewProps>(withScrollEnabler(forwardRef<Props>(FadedScrollView)), {
```

Comment:
This is a little out of scope. 
But what's nice about hooks (as long you work with function components) is that they are great alternative for HOC. 
For example in this case, you can create hooks for useScrollReached and useScrollEnabler that do calculation and return the relevant info which you can pass on to your ScrollView. 
Maybe it's a good opportunity to create an equivalent hooks for these HOCs

Reply:
Sounds great, however I'd rather do it in the second PR, so I won't have too many conflicts there (please add this comment there, so I don't forget 📝 ).

Reply:
coolio 👍